admin/class-boldgrid-backup-admin-core.php

```diff
@@ -2312,21 +2318,31 @@ public function restore_archive_file( $dryrun = false ) {
 			return [ 'error' => $error_message ];
 		}
 
-		// Initialize variables.
-		$archive_key      = null;
-		$archive_filename = null;
-
-		// Validate archive_key.
-		if ( isset( $_POST['archive_key'] ) && is_numeric( $_POST['archive_key'] ) ) { // phpcs:ignore WordPress.CSRF.NonceVerification.NoNonceVerification
+		/*
+		 * Get our archive key.
+		 *
+		 * It can be passed in via $args or $_POST.
+		 */
+		$archive_key = false;
+		if ( isset( $args['archive_key'] ) ) {
+			$archive_key = (int) $args['archive_key'];
+		} elseif ( isset( $_POST['archive_key'] ) && is_numeric( $_POST['archive_key'] ) ) { // phpcs:ignore WordPress.CSRF.NonceVerification.NoNonceVerification
 			$archive_key = (int) $_POST['archive_key'];
 		} else {
 			$error_message = esc_html__( 'Invalid key for the selected archive file.', 'boldgrid-backup' );
 			$this->logger->add( $error_message );
 			return [ 'error' => $error_message ];
 		}
 
-		// Validate archive_filename.
-		if ( ! empty( $_POST['archive_filename'] ) ) { // phpcs:ignore WordPress.CSRF.NonceVerification.NoNonceVerification
+		/*
+		 * Get our archive filename.
+		 *
+		 * It can be passed in via $args or $_POST.
+		 */
+		$archive_filename = false;
+		if ( ! empty( $args['archive_filename'] ) ) {
+			$archive_filename = sanitize_file_name( $args['archive_filename'] );
+		} elseif ( ! empty( $_POST['archive_filename'] ) ) { // phpcs:ignore WordPress.CSRF.NonceVerification.NoNonceVerification
```

Comment:
Fixed in https://github.com/BoldGrid/boldgrid-backup/commit/df3907dbcb69ca36d7400929466e68b08a1ebbb4

I suspect this may break a few more things... More testing needed.